Scripts/Tracking/Follow/ObjectFollow.cs

```diff
@@ -0,0 +1,217 @@
+                UpdatePosition(sourceTransform, targetTransform);
+
+                OnAfterTransformUpdated(sourceTransform, targetTransform);
+            }
+        }
+
+        /// <summary>
+        /// The UpdatePosition method executes the specified FollowModifier's UpdatePosition method.
+        /// </summary>
+        /// <param name="sourceTransform">The source Transform to apply the FollowModifier on.</param>
+        /// <param name="targetTransform">The target Transform to apply the FollowModifier with.</param>
+        protected virtual void UpdatePosition(Transform sourceTransform, Transform targetTransform)
+        {
+            if (follow.HasFlag(TransformProperties.Position))
+            {
+                OnBeforePositionUpdated(sourceTransform, targetTransform);
+                followModifier.UpdatePosition(sourceTransform, targetTransform);
+                OnAfterPositionUpdated(sourceTransform, targetTransform);
+            }
+        }
+
+        /// <summary>
+        /// The UpdateRotation method executes the specified FollowModifier's UpdateRotation method.
+        /// </summary>
+        /// <param name="sourceTransform">The source Transform to apply the FollowModifier on.</param>
+        /// <param name="targetTransform">The target Transform to apply the FollowModifier with.</param>
+        protected virtual void UpdateRotation(Transform sourceTransform, Transform targetTransform)
+        {
+            if (follow.HasFlag(TransformProperties.Rotation))
+            {
+                OnBeforeRotationUpdated(sourceTransform, targetTransform);
+                followModifier.UpdateRotation(sourceTransform, targetTransform);
+                OnAfterRotationUpdated(sourceTransform, targetTransform);
+            }
+        }
+
+        /// <summary>
+        /// The UpdateScale method executes the specified FollowModifier's UpdateScale method.
+        /// </summary>
+        /// <param name="sourceTransform">The source Transform to apply the FollowModifier on.</param>
+        /// <param name="targetTransform">The target Transform to apply the FollowModifier with.</param>
+        protected virtual void UpdateScale(Transform sourceTransform, Transform targetTransform)
+        {
+            if (follow.HasFlag(TransformProperties.Scale))
+            {
+                OnBeforeScaleUpdated(sourceTransform, targetTransform);
+                followModifier.UpdateScale(sourceTransform, targetTransform);
+                OnAfterScaleUpdated(sourceTransform, targetTransform);
+            }
+        }
+
+        /// <summary>
+        /// The GetTransformFromComponent method attempts to retrieve the Transform from a given Component.
+        /// </summary>
+        /// <param name="component">The Component to retrieve the Transform from.</param>
+        /// <returns>A Transform if one exists on the given Component.</returns>
+        protected virtual Transform GetTransformFromComponent(Component component)

```

Comment:
This is very generic and doesn't actually need anything from this particular class, nor can I see a need to be able to override this behavior in subclasses. Shall we move it to a `static` helper class instead?

We probably have this need often and sadly Unity's overriding of the `==` and `!=` operators mean we can't use C#'s `?` operator like `component?.transform` inline instead.

Reply:
Hmm it is generic but at the moment it's not being used anywhere else other than this file. Plus I'm not sure where else to put it, I don't want to end up having a Utility class that just has static methods in it again.

Reply:
What about an extension method on `Component`?

`ComponentExtensions.cs`:
```c#
public static class ComponentExtenstions
{
    public static Transform TryGetTransform(this Component component)
    {
        return component == null ? null : component.transform;
    }
}
```

Reply:
yeah I like `ComponentExtensions` idea. maybe have a Namespace of `Extension` perhaps in case we want to do additional extensions for other types?